Doc/library/asyncio-stream.rst

```diff
@@ -65,6 +66,14 @@ and work with streams:
    *limit* determines the buffer size limit used by the
    returned :class:`StreamReader` instance.  By default the *limit*
    is set to 64 KiB.
+   when the *limit* is 0 allocate a global buffer to read into.
+   By allocating a global buffer, each read call will read directly into the
```

Comment:
IMO this is a very low-level optimization for a very high-level (and frozen!) API. -1 on this.

Reply:
I think that the optimization allows for too good of a performance boost to pass on.
If you still think that this should be closed, then maybe the previous PR I made should be reverted too (https://github.com/python/cpython/pull/21442), as it allocates a buffer for each stream, that could blow up the memory on a server with a lot of clients.

Reply:
Ok I updated the PR to not have this API change

Reply:
> I think that the optimization allows for too good of a performance boost to pass on.

Have you benchmarked this specific optimization? If so please attach results, otherwise this is very speculative.

Reply:
I a did benchmark using the scripts I wrote at the description of this PR.
It's probably not the best benchmark ever but it does provide me some good results.
Also it's not that far fetched that not calling malloc / free each time you read a buffer improves performance.

Reply:
Huh, I didn't notice the benchmark results in the opening message.

>  It's probably not the best benchmark ever but it does provide me some good results.

You can potentially have a deque of 64kb buffers. Every stream could grab a buffer or allocate a new one if the deque is empty. The deque could have a max len set to 4 or some other relatively low number. That way you can have the benefits of global buffer and no visible API changes.

Reply:
I thought about that, the issue is that the only doing this right now is giving for example the first 4 protocols to be of type ``StreamReaderBufferedProtocol``, but there is no way to update that one these protocols got closed unless I implement __del__ on that class, which I don't think is the best way to do this.

If I'll come up with some better idea, I'll either update this PR or work on a new one.